tests/test_main.py

```diff
@@ -89,3 +89,37 @@ def _check(command: Command, path: t.Optional[t.List] = None):
     _check(ctx.command)
 
     assert len(errors) == 0, "\n".join(errors)
+
+
+@pytest.mark.skip  # TODO : unskip after resolving all options conflicts
+def test_if_there_are_no_option_duplicates(runner):
+    runner.invoke("--help")
+
+    ctx = app_context_holder.app_context
+    duplicates = {}
+
+    def _check(command: Command, path: t.Optional[t.List] = None):
+        path = path or ["snow"]
+
+        if duplicated_params := check_options_for_duplicates(command.params):
+            duplicates[" ".join(path)] = duplicated_params
+
+        if hasattr(command, "commands"):
+            for command_name, command_info in command.commands.items():
+                _check(command_info, [*path, command_name])
+
+    def check_options_for_duplicates(params: t.List[TyperOption]) -> t.Set[str]:
+        RESERVED_FLAGS = ["-h", "--help"]  # noqa: N806
```

Comment:
I'm not sure if `-h` should be reserved. 

It looks like Typer or click is handling the case well and removes `-h` from `--help` if a command requires `-h`.
Are you able to find any command which really has `-h` conflict?

However I see that having `-h` sometimes for `--help` and sometimes for sth else makes our options not consistent.

Reply:
Done